src/db/mod.rs

```diff
@@ -235,18 +235,25 @@ impl Times {
 #[derive(Debug, Default, PartialEq, Eq, Clone)]
 #[cfg_attr(feature = "serialization", derive(serde::Serialize))]
 pub struct CustomData {
-    pub items: Vec<CustomDataItem>,
+    pub items: HashMap<String, CustomDataItem>,
 }
 
-/// Custom data field for an entry or metadata
+/// Custom data field for an entry or metadata for internal use
 #[derive(Debug, Default, PartialEq, Eq, Clone)]
 #[cfg_attr(feature = "serialization", derive(serde::Serialize))]
 pub struct CustomDataItem {
-    pub key: String,
     pub value: Option<Value>,
     pub last_modification_time: Option<NaiveDateTime>,
 }
 
+/// Custom data field for an entry or metadata from XML data
+#[derive(Debug, Default, PartialEq, Eq, Clone)]
+#[cfg_attr(feature = "serialization", derive(serde::Serialize))]
+pub struct CustomDataItemFromXml {
```

Comment:
@varjolintu how about `CustomDataItemDenormalized`?

Reply:
Yeah the "From" word is not something I'd like to see in a serialization struct. Let's change that.